File: cm_custom/public/js/cscripts/sales_invoice.js
export default function sales_invoice() {
  return {
    is_pos: async function (frm) {
      const { is_pos, company } = frm.doc;
      if (is_pos && company && !frm.doc.pos_profile) {
        const { message: { branch } = {} } = await frappe.call({
          method: 'erpnext.stock.get_item_details.get_pos_profile',
          args: { company },
        });
        frm.set_value({ branch });
      }
    },
    pos_profile: async function (frm) {
      const { pos_profile } = frm.doc;
      if (pos_profile) {
        const { message: { branch } = {} } = await frappe.db.get_value(
          'POS Profile',
          pos_profile,
          'branch'
        );
        frm.set_value({ branch });
      }
    },
  };
}

File: cm_custom/doc_events/sales_invoice.py
# -*- coding: utf-8 -*-
from __future__ import unicode_literals
import frappe


def set_missing_values(doc, method, for_validate=False):
    doc.branch = frappe.get_cached_value("POS Profile", doc.pos_profile, "branch")




def before_save(doc, method):
    if not doc.is_pos:
        doc.payments = []
        if doc.is_return:
            doc.outstanding_amount = 0
